src/system_metrics/service_mongoose_system_metrics.erl

```diff
@@ -66,14 +67,17 @@ init(Args) ->
 
 handle_info(spawn_reporter, #system_metrics_state{report_after = ReportAfter,
                                                   reporter_monitor = none,
-                                                  reporter_pid = none} = State) ->
+                                                  reporter_pid = none,
+                                                  prev_report = PrevReport} = State) ->
+    ServicePid = self(),
     case get_client_id() of
         {ok, ClientId} ->
             {Pid, Monitor} = spawn_monitor(
                 fun() ->
-                    Reports = mongoose_system_metrics_collector:collect(),
+                    Reports = mongoose_system_metrics_collector:collect(PrevReport),
                     mongoose_system_metrics_sender:send(ClientId, Reports),
-                    mongoose_system_metrics_file:save(Reports)
+                    mongoose_system_metrics_file:save(Reports),
+                    ServicePid ! {prev_report, Reports}
```

Comment:
Actually, while this is perhaps more readable, it also means this new process sends two messages to the master process, one with this bang, and another one as the `'DOWN'` message when it dies. If here we do instead something like
```erlang
    exit({report, Report})
```
Only one message will be sent, with the Report, which can be caught in the `handle_info` as it is done already, with something like
```erlang
 handle_info({'DOWN', CollectorMonitor, _, _, {report, Report}}, 
```
Just saying, maybe is an option? (@michalwski)

Reply:
I see what you mean @NelsonVides. At this moment I'm not convinced sending a message with result via `exit` is clear. In my opinion it may lead to confusion.